src/common/transformations/src/transformations/common_optimizations/optimize_gather_nd.cpp

```diff
@@ -0,0 +1,90 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/common_optimizations/optimize_gather_nd.hpp"
+
+#include <memory>
+#include <ngraph/op/util/op_types.hpp>
+#include <ngraph/opsets/opset8.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+#include <ngraph/rt_info.hpp>
+#include <openvino/op/util/gather_nd_base.hpp>
+
+#include "itt.hpp"
+
+ov::pass::OptimizerGatherND::OptimizerGatherND() {
+    MATCHER_SCOPE(OptimizerGatherND);
+    auto gather_nd =
+        pattern::wrap_type<ov::op::util::GatherNDBase>({pattern::any_input(), pattern::wrap_type<op::v0::Constant>()});
+
+    matcher_pass_callback callback = [=](pattern::Matcher& m) {
+        auto gather_nd_node = std::dynamic_pointer_cast<ov::op::util::GatherNDBase>(m.get_match_root());
+        if (!gather_nd_node) {
+            return false;
+        }
```

Comment:
I haven't found check for GatherND `batch_dims` attribute, but the MO transformation is applied only to `batch_dims==0`.
https://github.com/openvinotoolkit/openvino/blob/72505b1d82f8782cacfa4740e17fb86629f80733/tools/mo/openvino/tools/mo/middle/GatherNdNormalizer.py#L38